src/fidesops/models/saasconfig.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Dict, List, Literal, Optional, Union
+from pydantic import BaseModel
+from fidesops.graph.config import Collection, FieldAddress, ScalarField
+
+from fidesops.graph.traversal import Dataset
+from fidesops.schemas.dataset import FidesopsDatasetReference
+from fidesops.schemas.shared_schemas import FidesOpsKey
+
+
+class ConnectorParams(BaseModel):
+    """
+    Required information for the given SaaS connector, can either be user provided or default to a given value.
+    Secret values should NOT be specified with a default_value
+    """
+
+    name: str
+    default_value: str
+    from_user: bool
+    is_secret: bool
+
+
+class RequestParam(BaseModel):
+    """
+    A request parameter which includes the type (query, path, or body) along with a default value or
+    a reference to an identity value or a value in another dataset
+    """
+
+    name: str
+    type: Literal[
+        "query", "path", "body"
+    ]  # used to determine location in the generated request
+    default_value: Optional[Any]
+    identity: Optional[str]
+    data_type: Optional[str]
+    references: Optional[List[FidesopsDatasetReference]]
+
+
+class Strategy(BaseModel):
+    """General shape for swappable strategies (ex: auth, pagination, postprocessing, etc.)"""
+
+    strategy_name: str
+    configuration: Dict[str, Any]
+
+
+class Request(BaseModel):
+    """
+    A single request with a static or dynamic path, and the request params needed to build the request.
+    Also includes optional startegies for pre/postprocessing and pagination
+    """
+
+    path: str
+    request_params: Optional[List[RequestParam]]
+    data_path: Optional[str]  # defaults to collection name if not specified
```

Comment:
`data_path` is the JSON path expression to be able to access nested data within a raw HTTP response. There are cases where a response will contain additional information. For example, the Mailchimp `/search-members` endpoint returns a response which we need to "unrwap" by calling `response["exact_matches"]["members"]`

Reply:
We call this same concept a `field_path`, which was added to access nested data, I was wondering if it might make sense to call it a field_path here.

Another question I have is do we need to handle the scenario where response_data can contain arrays?